src/github.com/matrix-org/dendrite/common/test/config.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2017 Vector Creations Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/pem"
+	"fmt"
+	"github.com/matrix-org/dendrite/common/config"
+	"github.com/matrix-org/gomatrixserverlib"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"math/big"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+const (
+	// ConfigFile is the name of the config file for a server.
+	ConfigFile = "dendrite.yaml"
+	// ServerKeyFile is the name of the file holding the matrix server private key.
+	ServerKeyFile = "server_key.pem"
+	// TLSCertFile is the name of the file holding the TLS certificate used for federation.
+	TLSCertFile = "tls_cert.pem"
+	// TLSKeyFile is the name of the file holding the TLS key used for federation.
+	TLSKeyFile = "tls_key.pem"
+	// MediaDir is the name of the directory used to store media.
+	MediaDir = "media"
+)
+
+// MakeConfig makes a config suitable for running integration tests.
+// Generates new matrix and TLS keys for the server.
+func MakeConfig(configDir, kafkaURI, database, host string, startPort int) (*config.Dendrite, int, error) {
+	var cfg config.Dendrite
+
+	port := startPort
+	assignAddress := func() config.Address {
+		result := config.Address(fmt.Sprintf("%s:%d", host, port))
+		port++
+		return result
+	}
+
+	serverKeyPath := filepath.Join(configDir, ServerKeyFile)
+	tlsCertPath := filepath.Join(configDir, TLSKeyFile)
+	tlsKeyPath := filepath.Join(configDir, TLSCertFile)
+	mediaBasePath := filepath.Join(configDir, MediaDir)
+
+	if err := newMatrixKey(serverKeyPath); err != nil {
+		return nil, 0, err
+	}
+
+	if err := newTLSKey(tlsKeyPath, tlsCertPath); err != nil {
+		return nil, 0, err
+	}
+
+	cfg.Version = config.Version
+
+	cfg.Matrix.ServerName = gomatrixserverlib.ServerName(assignAddress())
+	cfg.Matrix.PrivateKeyPath = config.Path(serverKeyPath)
+	cfg.Matrix.FederationCertificatePaths = []config.Path{config.Path(tlsCertPath)}
+
+	cfg.Media.BasePath = config.Path(mediaBasePath)
+
+	cfg.Kafka.Addresses = []string{kafkaURI}
+	// TODO: Different servers should be using different topics.
+	// Make this configurable somehow?
```

Comment:
I opted for pre-/post-fixes in the media api integration tests. Maybe something like that?